pkg/generic/thrift/json_go116plus_amd64.go

```diff
@@ -0,0 +1,156 @@
+//go:build amd64 && go1.16
+// +build amd64,go1.16
+
+/*
+ * Copyright 2023 CloudWeGo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package thrift
+
+import (
+	"context"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/bytedance/gopkg/lang/mcache"
+	"github.com/cloudwego/dynamicgo/conv/j2t"
+
+	"github.com/cloudwego/kitex/pkg/generic/descriptor"
+	"github.com/cloudwego/kitex/pkg/remote/codec/perrors"
+	cthrift "github.com/cloudwego/kitex/pkg/remote/codec/thrift"
+	"github.com/cloudwego/kitex/pkg/utils"
+)
+
+// Write write json string to out thrift.TProtocol
+func (m *WriteJSON) Write(ctx context.Context, out thrift.TProtocol, msg interface{}, requestBase *Base) error {
+	// fallback logic
+	if !m.dynamicgoEnabled {
+		return m.originalWrite(ctx, out, msg, requestBase)
+	}
+
+	var cv j2t.BinaryConv
+	// dynamicgo logic
+	if !m.hasRequestBase {
+		requestBase = nil
+	}
+	if requestBase != nil {
+		cv = j2t.NewBinaryConv(*m.dynamicgoConvOptsWithThriftBase)
+	} else {
+		cv = j2t.NewBinaryConv(*m.dynamicgoConvOpts)
+	}
+
+	// msg is void
+	if _, ok := msg.(descriptor.Void); ok {
+		if err := m.writeHead(out); err != nil {
+			return err
+		}
+		if err := m.writeFields(ctx, out, Void, nil, nil, nil); err != nil {
+			return err
+		}
+		return writeTail(out)
+	}
+
+	// msg is string
+	s, ok := msg.(string)
+	if !ok {
+		return perrors.NewProtocolErrorWithType(perrors.InvalidData, "decode msg failed, is not string")
+	}
+
+	transBuff := utils.StringToSliceByte(s)
+	dbuf := mcache.Malloc(len(transBuff))[0:0]
```

Comment:
btw, is there some code responsible for calling `mache.Free(dbuf)`?

Reply:
@felix021 Just the code below?

Reply:
@felix021 maybe you can refresh the page

Reply:
I missed it.

If you're sure the memory allocated to dbuf will not be refered for this request

Reply:
@felix021 Yeah, I think dbuf will not be refered after the function. (But for read function, I should not use mcache.Free, so I'll fix)